crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/meta_net_task.rs

```diff
@@ -0,0 +1,1157 @@
+            .block(Block::new(
+                BlockTarget::Node(test_node_id(1), NodeBlockReason::DOS),
+                InclusiveTimestampInterval::try_new(
+                    Timestamp::now(),
+                    Timestamp::now()
+                        .checked_add(&Duration::from_secs(10))
+                        .unwrap(),
+                )
+                .unwrap(),
+            ))
+            .await
+            .unwrap();
+
+        let con = mk_test_con_with_id(1);
+        let con_state = get_con_state(&con);
+
+        ep_evt_send
+            .send(MetaNetEvt::Request {
+                remote_url: "".to_string(),
+                con: con,
+                data: wire::Wire::Call(wire::Call {
+                    space: test_space(1),
+                    to_agent: test_agent(2),
+                    data: WireData(vec![]),
+                }),
+                respond: Box::new(|_| async move { () }.boxed().into()),
+            })
+            .await
+            .unwrap();
+
+        tokio::time::timeout(Duration::from_millis(1000), async {
+            while !con_state.read().closed {
+                tokio::time::sleep(Duration::from_millis(1)).await;
+            }
+        })
+        .await
+        .expect("Timed out waiting for the connection to be closed");
+
+        assert!(con_state.read().closed);
+    }
+
+    #[tokio::test(flavor = "multi_thread")]
+    async fn make_call_request() {
+        let (mut ep_evt_send, _, _, _, _, _) = setup().await;
+
+        let (send_res, read_res) = futures::channel::oneshot::channel();
+
+        let request_data = vec![2, 7];
+        ep_evt_send
+            .send(MetaNetEvt::Request {
+                remote_url: "".to_string(),
+                con: mk_test_con_with_id(1),
+                data: wire::Wire::Call(wire::Call {
+                    space: test_space(1),
+                    to_agent: test_agent(2),
+                    data: WireData(request_data.clone()),
+                }),
+                respond: Box::new(|r| {
+                    async move {
+                        send_res.send(r).unwrap();
+                        ()
+                    }
+                    .boxed()
+                    .into()
+                }),
+            })
+            .await
+            .unwrap();
+
+        let call_response = tokio::time::timeout(Duration::from_secs(1), read_res)
+            .await
+            .expect("Timed out while waiting for a response")
+            .unwrap();
+
+        let response_data = match call_response {
+            Wire::CallResp(res) => res.data.to_vec(),
+            _ => panic!("Unexpected response"),
+        };
+
+        // Because the stub does an echo response
+        assert_eq!(request_data, response_data);
+    }
+
+    // TODO This is actually a fatal error so the task should stop and then I need another mechanism to cause an error
+    //      to test the behaviour this test is currently checking.
```

Comment:
I actually have this mechanism now, I just need to make the change. Showing the code/tests as they are because this is already a complex diff. I'd rather make this change on a next PR.